### PR TITLE
CAT-277 Move NestedBrick

### PR DESCRIPTION
**Pull request by bernadettespieler**
such as if-Brick, Forever and Repeat _ times

### CODE REVIEW

File: catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java [hunk 1]
```diff
@@ -298,6 +302,16 @@ private int getNewPositionIfEndingBrickIsThere(int to, Brick brick) {
 		return to;
 	}
 
+	private int adjustNestingBrickDraggedPosition(NestingBrick nestingBrick, int from, int to) {
+		List<NestingBrick> nestingBrickList = nestingBrick.getAllNestingBrickParts(true);
+		NestingBrick endBrick = nestingBrickList.get(nestingBrickList.size() - 1);
+		int endBrickPosition =  brickList.indexOf(endBrick);
+		if (to > from && to < endBrickPosition) {
```
Comment (thomasmauerhofer) [inline catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java]: Please use an boolean value with a meaningful name

File: catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java [hunk 2]
```diff
@@ -475,6 +493,48 @@ private void moveUserBrick(int from, int to) {
...
 
+	private void moveNestingBrick(int from, int to) {
+		Sprite currentSprite = ProjectManager.getInstance().getCurrentSprite();
+
+		int[] tempFrom = getScriptAndBrickIndexFromProject(from);
+		int scriptPositionFrom = tempFrom[0];
+		int brickPositionFrom = tempFrom[1];
+
+		Script fromScript = currentSprite.getScript(scriptPositionFrom);
+
```
Comment (thomasmauerhofer) [inline catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java]: remove newline

File: catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java [hunk 3]
```diff
@@ -475,6 +493,48 @@ private void moveUserBrick(int from, int to) {
...
+		Brick brick = fromScript.getBrick(brickPositionFrom);
+
+		NestingBrick nestingBrick = (NestingBrick) brick;
+		List<NestingBrick> nestingBricks = nestingBrick.getAllNestingBrickParts(true);
+		NestingBrick endNestingBrick = nestingBricks.get(nestingBricks.size() - 1);
+		List<Brick> fromScriptBrickList = fromScript.getBrickList();
+		int endPosition = fromScriptBrickList.indexOf(endNestingBrick);
+		int count = endPosition - brickPositionFrom;
+		if (to >  from && count > to - from) {
```
Comment (thomasmauerhofer) [inline catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java]: Please use an boolean value with a meaningful name

File: catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java [hunk 4]
```diff
@@ -475,6 +493,48 @@ private void moveUserBrick(int from, int to) {
...
+		removedBlock.add(block.remove(0));
+
+		int[] tempTo = getScriptAndBrickIndexFromProject(to);
+		int scriptPositionTo = tempTo[0];
+		int brickPositionTo = tempTo[1];
+
+		Script toScript = currentSprite.getScript(scriptPositionTo);
+
+		while (block.size() > 0) {
+			removedBlock.add(block.remove(0));
```
Comment (thomasmauerhofer) [inline catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java]: Isn't it easier to use addAll()?

File: catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java [hunk 5]
```diff
@@ -475,6 +493,48 @@ private void moveUserBrick(int from, int to) {
...
+		}
+
+		int finalBrickPositionTo = brickPositionTo;
+		if (scriptPositionTo == scriptPositionFrom && to > from) {
```
Comment (thomasmauerhofer) [inline catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java]: Please use an boolean value with a meaningful name

File: catroidTest/src/org/catrobat/catroid/uitest/ui/fragment/ScriptFragmentTest.java [hunk 6]
```diff
@@ -422,6 +422,25 @@ public void testSimpleDragNDrop() {
 		assertEquals("Incorrect brick order after dragging & dropping", brickListToCheck.get(4), brickList.get(4));
 	}
 
+	public void testDragNDropNestedBrick() {
+		List<Brick> brickListToCheck = UiTestUtils.createTestProjectNestedBricks();
+		UiTestUtils.getIntoScriptActivityFromMainMenu(solo);
+		ArrayList<Integer> yPositionList = UiTestUtils.getListItemYPositions(solo, 0);
+		assertTrue("Test project brick list smaller than expected", yPositionList.size() >= 8);
```
Comment (thomasmauerhofer) [inline catroidTest/src/org/catrobat/catroid/uitest/ui/fragment/ScriptFragmentTest.java]: use ==

File: catroidTest/src/org/catrobat/catroid/uitest/util/UiTestUtils.java [hunk 7]
```diff
@@ -1132,6 +1132,46 @@ public static void createTestProjectWithNestedUserBrick() {
 		return brickList;
 	}
 
+	public static List<Brick> createTestProjectNestedBricks() {
+		Project project = new Project(null, DEFAULT_TEST_PROJECT_NAME);
+		Sprite firstSprite = new Sprite("cat");
+
```
Comment (thomasmauerhofer) [inline catroidTest/src/org/catrobat/catroid/uitest/util/UiTestUtils.java]: remove newline

File: catroidTest/src/org/catrobat/catroid/uitest/util/UiTestUtils.java [hunk 8]
```diff
@@ -1132,6 +1132,46 @@ public static void createTestProjectWithNestedUserBrick() {
...
+		Script testScript = new StartScript();
+
```
Comment (thomasmauerhofer) [inline catroidTest/src/org/catrobat/catroid/uitest/util/UiTestUtils.java]: remove newline

File: catroidTest/src/org/catrobat/catroid/uitest/util/UiTestUtils.java [hunk 9]
```diff
@@ -1132,6 +1132,46 @@ public static void createTestProjectWithNestedUserBrick() {
...
+		brickList.add(repeatBrick);
+		brickList.add(new ShowBrick());
+		brickList.add(loopEndBrick);
+
+		for (Brick brick : brickList) {
+			testScript.addBrick(brick);
+		}
+
+		firstSprite.addScript(testScript);
+
```
Comment (thomasmauerhofer) [inline catroidTest/src/org/catrobat/catroid/uitest/util/UiTestUtils.java]: remove newline

File: catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java [hunk 10]
```diff
@@ -475,6 +495,47 @@ private void moveUserBrick(int from, int to) {
...
+		Script fromScript = currentSprite.getScript(scriptPositionFrom);
+		Brick brick = fromScript.getBrick(brickPositionFrom);
+
+		NestingBrick nestingBrick = (NestingBrick) brick;
+		List<NestingBrick> nestingBricks = nestingBrick.getAllNestingBrickParts(true);
+		NestingBrick endNestingBrick = nestingBricks.get(nestingBricks.size() - 1);
+		List<Brick> fromScriptBrickList = fromScript.getBrickList();
+		int endPosition = fromScriptBrickList.indexOf(endNestingBrick);
+		int count = endPosition - brickPositionFrom;
+		boolean isNewPositionBetweenStartAndEndNestedBrick = to >  from && count > to - from;
```
Comment (thmq) [inline catroid/src/org/catrobat/catroid/ui/adapter/BrickAdapter.java]: please remove the unnecessary space after "to > " there should be only one. (There is one more instance of the same problem -> just have a look at the formatters suggestions.

**Pull request closed without merging**